### PR TITLE
Create automated bugreporter.

### PR DESCRIPTION
**Pull request by dundargoc**
Using the issue template as a starting point. 

- [x] - `nvim --version`:

- [x] - Operating system/version:
Currently use vim.loop.os_uname(), thanks to a review from @mjlbach. Not sure if this is reliable enough to keep it or if I need to go full neofetch like @sunjon suggested. 

- [ ] - Terminal name/version:
Don't think there's a consistent way to get this information, I think I'll need to rely on different values of TERM and god know what else environment variables.

- [x] - `$TERM`:

### CODE REVIEW

File: runtime/lua/bugreport.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
+-- Copied and modified gist.github.com/soulik/82e9d02a818ce12498d1
+local function getOS()
+	local raw_os_name = ""
+
+	-- LuaJIT shortcut
+	if jit and jit.os then
+		raw_os_name = jit.os
+	else
+		-- is popen supported?
+		local popen_status, popen_result = pcall(io.popen, "")
```
Comment (bfredl) [inline runtime/lua/bugreport.lua:10]: You could check windows using `vim.fn.has("win32") ~= 0`

**Comment by sunjon** [comment]
NeoFetch produces the same kind of info you're looking for (and is battle tested to the tune of 12k stars).
You might want to consider using their [strategies for terminal](https://github.com/dylanaraps/neofetch/blob/7913ad35ffb5f76bec0b961d4ba94f3ad2eddf4a/neofetch#L3281) and OS+distro+version detection.

File: runtime/lua/bugreport.lua (hunk 1 above)
Comment (dundargoc) [reply runtime/lua/bugreport.lua]: I think I changed it like you meant, but I honestly have no idea what I'm doing.

File: runtime/lua/bugreport.lua [hunk 2]
```diff
@@ -0,0 +1,70 @@
+-- Copied and modified gist.github.com/soulik/82e9d02a818ce12498d1
+local function getOS()
+	local raw_os_name = ""
+
+	-- LuaJIT shortcut
+	if jit and jit.os then
```
Comment (mjlbach) [inline runtime/lua/bugreport.lua:6]: You can also match on os_uname

Comment (dundargoc) [reply runtime/lua/bugreport.lua]: Well, that was certainly useful. Thanks for the help!

File: runtime/lua/bugreport.lua [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	local NVIM_VERSION = fn.execute("version")
+	local SYSTEM_INFO = vim.loop.os_uname()
+	local DISTRO_INFO = fetch_distro_info()
+
+	template_pre_filled = TEMPLATE:format(
+		NVIM_VERSION,
+		SYSTEM_INFO.sysname,
+		SYSTEM_INFO.release,
+		SYSTEM_INFO.version,
+		DISTRO_INFO.PRETTY_NAME or (DISTRO_INFO.NAME .. " " .. DISTRO_INFO.VERSION) or "",
```
Comment (nanotee) [inline runtime/lua/bugreport.lua:61]: Just realized this is a potential bug: if either `DISTRO_INFO.NAME` or `DISTRO_INFO.VERSION` is `nil`, the code will fail.
```suggestion
		DISTRO_INFO.PRETTY_NAME or ((DISTRO_INFO.NAME or "") .. " " .. (DISTRO_INFO.VERSION or "")),
```

**Comment by clason** [comment]
Note to self: this could be integrated into a `checkhealth` revamp, which should contain more diagnostic output that would be relevant to these reports. 

(For example, rather than just "no issues found" -- what does that mean? what "issues" have been checked? -- it could print the name and location of the `init` file (esp. vim vs. lua), the full `rtp`/`packpath`, list of packages/modules loaded etc.)

**Comment by dundargoc** [comment]
Closing as I believe this will create more work than it will save in its current state. Terminals change and new ones are created constantly. Keeping up with this change for something that the user can provide with better accuracy seems counterproductive.

**Pull request closed without merging**